edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -321,6 +324,68 @@ impl ModuleRuntime for DockerModuleRuntime {
         Box::new(result)
     }
 
+    fn get(&self, id: &str) -> Self::GetFuture {
+        fn parse_response<'de, D>(resp: &InlineResponse200) -> std::result::Result<String, D::Error>
+        where
+            D: serde::Deserializer<'de>,
+        {
+            let name = resp
+                .name()
+                .map(ToOwned::to_owned)
+                .ok_or_else(|| serde::de::Error::missing_field("Name"))?;
+            Ok(name)
+        }
+
+        debug!("Getting module {}...", id);
+
+        let id = id.to_string();
+
+        if let Err(err) = ensure_not_empty_with_context(&id, || {
+            ErrorKind::RuntimeOperation(RuntimeOperation::GetModule(id.clone()))
+        }) {
+            return Box::new(future::err(Error::from(err)));
+        }
+
+        let client_copy = self.client.clone();
+
+        Box::new(
+            self.client
+                .container_api()
+                .container_inspect(&id, false)
+                .then(|result| match result {
+                    Ok(container) => {
+                        let name = parse_response::<
+                            &mut serde_json::Deserializer<serde_json::de::IoRead<std::io::Empty>>,
+                        >(&container)
+                        .with_context(|_| {
+                            ErrorKind::RuntimeOperation(RuntimeOperation::GetModule(id.clone()))
+                        })?;
+                        let config =
+                            DockerConfig::new(name.clone(), ContainerCreateBody::new(), None)
+                                .with_context(|_| {
+                                    ErrorKind::RuntimeOperation(RuntimeOperation::GetModule(
+                                        id.clone(),
+                                    ))
+                                })?;
+                        let module =
+                            DockerModule::new(client_copy, name, config).with_context(|_| {
+                                ErrorKind::RuntimeOperation(RuntimeOperation::GetModule(id.clone()))
+                            })?;
+                        let state = runtime_state(container.id(), container.state());
```

Comment:
I see what you are doing here, but the API seems a little odd. Now I can get the same information using Module.GetRuntimeState() or Runtime.Get(moduleId). Would be good to see if we can make this unified somehow...

Reply:
Agreed. At least they share common internal code for now (the `runtime_state` function). I just didn't want my changes to ripple out too far all at once.

Reply:
BTW my first thought was to use `ModuleRuntime::get` to return the `Module`, and from there call `Module::runtime_state`. But the problem is that `docker inspect` returns information about a module _and_ its runtime state together, so the above flow would result in an extra, unnecessary call to `docker inspect`.

Reply:
Yes, I understand that.. 
One way I could think of was - Module can have runtime_state from before (when we call ModuleRuntime.Get()), or not (when it is created as part of List). If it does, Module::runtime_state just returns it, if it doesn't, it calls inspect. 

Another way would be for RuntimeState to be a top level thing, which contains Module, instead of the other way round. I think that might be cleanest..